src/library/scala/collection/mutable/SortedMap.scala

```diff
@@ -0,0 +1,53 @@
+package scala
+package collection
+package mutable
+
+import generic._
+
+/**
+ * A mutable map whose keys are sorted.
+ *
+ * @tparam A the type of the keys contained in this sorted map.
+ * @tparam B the type of the values associated with the keys.
+ *
+ * @author Rui Gonçalves
+ * @version 2.12
+ * @since 2.12
+ *
+ * @define Coll mutable.SortedMap
+ * @define coll mutable sorted map
+ */
+trait SortedMap[A, B]
+  extends Map[A, B]
+  with collection.SortedMap[A, B]
+  with MapLike[A, B, SortedMap[A, B]]
+  with SortedMapLike[A, B, SortedMap[A, B]] {
+
+  override protected[this] def newBuilder: Builder[(A, B), SortedMap[A, B]] = SortedMap.newBuilder[A, B]
+
+  override def empty: SortedMap[A, B] = SortedMap.empty
+  override def updated[B1 >: B](key: A, value: B1): SortedMap[A, B1] = this + ((key, value))
+
+  override def +[B1 >: B](kv: (A, B1)): SortedMap[A, B1] = clone().asInstanceOf[SortedMap[A, B1]] += kv
+
+  override def +[B1 >: B](elem1: (A, B1), elem2: (A, B1), elems: (A, B1) *): SortedMap[A, B1] =
+    this + elem1 + elem2 ++ elems
+
+  override def ++[B1 >: B](xs: GenTraversableOnce[(A, B1)]): SortedMap[A, B1] =
+    clone().asInstanceOf[SortedMap[A, B1]] ++= xs.seq
+}
+
```

Comment:
You might want to also instantiate an abstract class for the `SortedMap` trait - see example here:

https://github.com/scala/scala/blob/v2.11.5/src/library/scala/collection/mutable/Map.scala#L78